src/test/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ImmutableComputableGraphUnitTest.java

```diff
@@ -0,0 +1,81 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.cachemanager;
+
+import junit.framework.AssertionFailedError;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Unit tests for {@link ImmutableComputableGraph}
+ *
+ * TODO github/gatk-protected issue #803
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public class ImmutableComputableGraphUnitTest extends BaseTest {
+
+    public static Function<Map<String, ? extends Duplicable>, ? extends Duplicable> func_0 = col -> {
+        final INDArray x = DuplicableNDArray.of(col.get("X"));
+        final double y = DuplicableNumber.of(col.get("Y"));
+        return new DuplicableNDArray(x.mul(y));
+    };
+
+    public static Function<Map<String, ? extends Duplicable>, ? extends Duplicable> func_1 = col -> {
+        final INDArray xProdY = DuplicableNDArray.of(col.get("X_prod_Y"));
+        final double y = DuplicableNumber.of(col.get("Y"));
+        return new DuplicableNDArray(xProdY.add(y));
+    };
+
```

Comment:
Skip tests.  I think these sorts of tests of the basic infrastructure definitely need to be in before a release, just to make sure no bugs are slipping by.  How long do you think it will take to write them?